virtcontainers/persist.go

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2018 Huawei Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package virtcontainers
+
+import (
+	"errors"
+
+	"github.com/kata-containers/runtime/virtcontainers/device/api"
+	exp "github.com/kata-containers/runtime/virtcontainers/experimental"
+	"github.com/kata-containers/runtime/virtcontainers/persist"
+	persistapi "github.com/kata-containers/runtime/virtcontainers/persist/api"
+	"github.com/kata-containers/runtime/virtcontainers/types"
+)
+
+var (
+	errContainerPersistNotExist = errors.New("container doesn't exist in persist data")
+)
+
+func (s *Sandbox) dumpState(ss *persistapi.SandboxState, cs map[string]persistapi.ContainerState) error {
```

Comment:
Global comment for this file:
Having new `Sandbox` methods out of the `sandbox.go` file sounds pretty confusing to me.

Reply:
I can move them to `sandbox.go, I am worried that `sandbox.go` is already so long, the single file has over 1800 lines, that's why I put the store part to one single file.

I can merge this file into sandbox.go in a following PR  if you insist .

Reply:
Ok course you can do that in a follow up PR :)
But I'd like to have input from @bergwolf @amshinde on this too.

Reply:
ping @bergwolf @amshinde again, let's finish the discussion :)